src/templates/calender/hooks/useCalender.tsx

```diff
@@ -0,0 +1,93 @@
+import DATE from "@/constants/date.constant";
+import dayjs from "dayjs";
+import React from "react";
+import useModal from "@/hooks/useModal";
+import useUser from "@/hooks/useUser";
+import { toast } from "react-toastify";
+import {
+  useAddCalenderPlanMutation,
+  useDeleteCalenderPlanMutation,
+} from "../services/mutation.service";
+import CalenderPlanAddModal from "../layouts/CalenderPlanAddModal";
+import { ChangeDateType } from "../types";
+import { CalenderPlanAddQuery } from "../interfaces";
+import { getPlanTypeByPlanName } from "../helpers";
+
+const useCalender = () => {
+  const [currentMonth, setCurrentMonth] = React.useState(dayjs().month());
+  const { openModal } = useModal();
+  const { user } = useUser();
+  const { mutate: deleteMutate } = useDeleteCalenderPlanMutation();
+  const { mutate: addPlanMutate } = useAddCalenderPlanMutation();
+
+  const firstDayOfMonth = () => {
+    const dayOfWeek = dayjs().year(dayjs().year()).month(currentMonth).date(1);
+    return Array.from({ length: dayOfWeek.day() });
+  };
+
+  const isPlanWriterIsUser = (id: number) => {
```

Comment:
is가 두 개 들어가서 외국인이 봤을 때 약간 어려움이 있을 것 같아요.
```suggestion
  const isPlanWriterSameAsUser = (id: number) => {
```

라고 짓는 건 어떨까요?

Reply:
좋은것같습니다!